sync-core/src/test/java/com/cloudant/sync/replication/ResponseStreamReaderFilter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.replication;
+
+import com.cloudant.http.HttpConnection;
+import com.cloudant.http.HttpConnectionFilterContext;
+import com.cloudant.http.HttpConnectionResponseFilter;
+
+import java.io.IOException;
+
+/**
+ * Created by Rhys Short on 20/07/15.
+ */
+public class ResponseStreamReaderFilter implements HttpConnectionResponseFilter {
```

Comment:
Some documentation might be useful here. It looks like you are reading the response stream which will cause the caller to error because the stream has been exhausted at the point it wants to read the response.

I'm not sure if this is a useful test and as discussed, it might highlight the fact that response filters are fairly useless in their current guise. Maybe it's worth leaving out anything to do with response filters in this PR until they can do useful things like replace the response stream etc